File: server-go/internal/domain/scm/orderflow/meta.go
package orderflow

import "github.com/z1coyan/synie/server/internal/platform/meta"

const ResourceName = "scmOrderFlowItems"

var flowTypeOptions = []meta.EnumOption{
	{Value: "PURCHASE_RECEIPT", Label: "采购入库"},
	{Value: "OUTSOURCED_ISSUE", Label: "委外发料"},
	{Value: "OUTSOURCED_RECEIPT", Label: "委外入库"},
	{Value: "SALES_DELIVERY", Label: "销售发货"},
}

var statusOptions = []meta.EnumOption{
	{Value: "DRAFT", Label: "草稿"},
	{Value: "AUDITED", Label: "已审核"},
	{Value: "VOIDED", Label: "已作废"},
}

func ResourceMeta() meta.ResourceMeta {
	return meta.ResourceMeta{
		Name: ResourceName, PermissionPrefix: "scm.order_flow",
		PermissionLabel: "订单收发货历史", Table: "scm_order_flow_item",
		ReadPermissionsAny: append([]string(nil), sourceReadPermissions...),
		Fields: []meta.FieldMeta{
			{Name: "id", APIName: "id", DBColumn: "id", Type: meta.TypeString, Label: "行标识(单据类型:来源行 id)", Readonly: true, Filterable: true, Sortable: true},
			{Name: "flow_type", APIName: "flowType", DBColumn: "flow_type", Type: meta.TypeEnum, Label: "单据类型", Readonly: true, EnumOptions: flowTypeOptions, Filterable: true, Sortable: true},
			{Name: "voucher_no", APIName: "voucherNo", DBColumn: "voucher_no", Type: meta.TypeString, Label: "单据编号", Readonly: true, Filterable: true, Sortable: true},
			{Name: "voucher_date", APIName: "voucherDate", DBColumn: "voucher_date", Type: meta.TypeDate, Label: "单据日期", Readonly: true, Filterable: true, Sortable: true},
			{Name: "status", APIName: "status", DBColumn: "status", Type: meta.TypeEnum, Label: "单据状态", Readonly: true, EnumOptions: statusOptions, Filterable: true, Sortable: true},
			{Name: "qty", APIName: "qty", DBColumn: "qty", Type: meta.TypeDecimal, Label: "数量", Readonly: true, Filterable: true, Sortable: true},
			{Name: "material_code", APIName: "materialCode", DBColumn: "material_code", Type: meta.TypeString, Label: "物料编号", Readonly: true, Filterable: true, Sortable: true},
			{Name: "material_name", APIName: "materialName", DBColumn: "material_name", Type: meta.TypeString, Label: "物料名称", Readonly: true, Filterable: true, Sortable: true},
			{Name: "material_spec", APIName: "materialSpec", DBColumn: "material_spec", Type: meta.TypeString, Label: "规格", Readonly: true, Filterable: true, Sortable: true},
			{Name: "customer_part_no", APIName: "customerPartNo", DBColumn: "customer_part_no", Type: meta.TypeString, Label: "客户料号", Readonly: true, Filterable: true, Sortable: true},
			{Name: "unit_name", APIName: "unitName", DBColumn: "unit_name", Type: meta.TypeString, Label: "单位名称", Readonly: true, Filterable: true, Sortable: true},
			{Name: "order_id", APIName: "orderId", DBColumn: "order_id", Type: meta.TypeUUID, Label: "订单", Readonly: true, Sortable: true},
			{Name: "order_item_id", APIName: "orderItemId", DBColumn: "order_item_id", Type: meta.TypeUUID, Label: "订单条目", Readonly: true, Sortable: true},
			{Name: "company_id", APIName: "companyId", DBColumn: "company_id", Type: meta.TypeUUID, Label: "公司", Readonly: true, Sortable: true},
		},
		// 权限完全复用四种来源单据的 read OR，不进入权限目录。
		Actions: nil,
		Audit:   meta.AuditMeta{Enabled: false},
	}
}

File: server/src/modules/scm/orderflow/meta.ts
import type { ResourceMeta } from '~/platform/meta/types.ts'

export const ORDER_FLOW_SOURCE_READ_PERMISSIONS = [
  'purchase.receipt:read',
  'purchase.outsourced_issue:read',
  'purchase.outsourced_receipt:read',
  'sales.delivery:read',
  'sales.return:read',
  'purchase.return:read',
  'purchase.outsourced_return:read',
] as const

const FLOW_TYPES = [
  { value: 'PURCHASE_RECEIPT', label: '采购入库' },
  { value: 'OUTSOURCED_ISSUE', label: '委外发料' },
  { value: 'OUTSOURCED_RECEIPT', label: '委外入库' },
  { value: 'SALES_DELIVERY', label: '销售发货' },
  { value: 'SALES_RETURN', label: '销售退货' },
  { value: 'PURCHASE_RETURN', label: '采购退货' },
  { value: 'OUTSOURCED_RETURN', label: '委外退货' },
]

const STATUS = [
  { value: 'DRAFT', label: '草稿' },
  { value: 'AUDITED', label: '已审核' },
  { value: 'VOIDED', label: '已作废' },
]

function f(
  name: string,
  apiName: string,
  type: ResourceMeta['fields'][number]['type'],
  label: string,
  opts: Partial<ResourceMeta['fields'][number]> = {},
): ResourceMeta['fields'][number] {
  return { name, apiName, dbColumn: name, type, label, readonly: true, ...opts }
}

export function orderFlowItemMeta(): ResourceMeta {
  return {
    name: 'scmOrderFlowItems',
    classification: { presentation: 'none', interactive: false, note: '订单流只读投影' },
    permissionPrefix: 'base.order_flow',
    permissionLabel: '订单收发货历史',
    table: 'scm_order_flow_item',
    authz: { kind: 'company', readAnyOf: [...ORDER_FLOW_SOURCE_READ_PERMISSIONS] },
    fields: [
      f('id', 'id', 'string', '行标识(单据类型:来源行 id)', {
        filterable: true,
        sortable: true,
      }),
      f('flow_type', 'flowType', 'enum', '单据类型', {
        enumOptions: FLOW_TYPES,
        filterable: true,
        sortable: true,
      }),
      f('voucher_no', 'voucherNo', 'string', '单据编号', {
        filterable: true,
        sortable: true,
      }),
      f('voucher_date', 'voucherDate', 'date', '单据日期', {
        filterable: true,
        sortable: true,
      }),
      f('status', 'status', 'enum', '单据状态', {
        enumOptions: STATUS,
        filterable: true,
        sortable: true,
      }),
      f('qty', 'qty', 'decimal', '数量', { filterable: true, sortable: true }),
      f('material_code', 'materialCode', 'string', '物料编号', {
        filterable: true,
        sortable: true,
      }),
      f('material_name', 'materialName', 'string', '物料名称', {
        filterable: true,
        sortable: true,
      }),
      f('material_spec', 'materialSpec', 'string', '规格', {
        filterable: true,
        sortable: true,
      }),
      f('customer_part_no', 'customerPartNo', 'string', '客户料号', {
        filterable: true,
        sortable: true,
      }),
      f('unit_name', 'unitName', 'string', '单位名称', {
        filterable: true,
        sortable: true,
      }),
      f('order_id', 'orderId', 'uuid', '订单', { sortable: true }),
      f('order_item_id', 'orderItemId', 'uuid', '订单条目', { sortable: true }),
      f('company_id', 'companyId', 'uuid', '公司', { sortable: true }),
    ],
    actions: [],
    audit: { enabled: false },
  }
}
